src/tcgwars/logic/impl/gen3/Deoxys.groovy

```diff
@@ -931,15 +948,7 @@ public enum Deoxys implements LogicCardInfo {
             energyCost D
             attackRequirement {}
             onAttack {
-              delayed {
-                before PLAY_TRAINER, {
-                  if (ef.cardToPlay.cardTypes.is(SUPPORTER) && bg.currentTurn == self.owner.opposite) {
-                    wcu "Limitation prevents playing this card"
-                    prevent()
-                  }
-                }
-                unregisterAfter 2
-              }
+              opponentCantPlaySupporterNextTurn(delegate)
```

Comment:
I've got concerns about this method actually.
1. It doesn't check if it was played from hand, but I'm not sure that is an issue.
1. It explicitly says "**Bother** prevents playing supporters" 
1. Looking at it, it unregisters if this Pokémon (Sableye here) Evolves or Switches. I'm not sure evolve even matters considering I have no knowledge of the opponent being able to evolve your Pokémon, but is it actually supposed to unregister on Sableye switching?

Reply:
> It doesn't check if it was played from hand, but I'm not sure that is an issue.

Might need to be tested against Impersonate, Download or Telepass. But those shouldn't be detected as supporters in the first place.

> It explicitly says "Bother prevents playing supporters"

Yeah, that needs a fix so other attacks can use it (same with GRI's Sableye). But probably not the PR to put it in. Will put a TODO as a reminder.

> Looking at it, it unregisters if this Pokémon (Sableye here) Evolves or Switches. I'm not sure evolve even matters considering I have no knowledge of the opponent being able to evolve your Pokémon, but is it actually supposed to unregister on Sableye switching?

Again, agree but should be fixed on a specific statics edit. It's an effect on the player so should remain even if Sableye is switched/evolved(?)/KO'd.